From 51ffa456864a588f51f97d071b77e899698507fc Mon Sep 17 00:00:00 2001
From: Chih-Wei Huang <cwhuang@linux.org.tw>
Date: Fri, 23 Sep 2016 17:59:47 +0800
Subject: [PATCH 11/24] init: set default value of ro.hardware to the target
 name

People usually forget to set the value of androidboot.hardware to
the target name in kernel cmdline which results in boot failure.

The change tries to reduce such an error in the future. The target
name is hardcoded to the image. This eliminates the necessity of
androidboot.hardware.

Conflicts:
	init/Android.mk

Change-Id: Ieef00dce33f52e556bad66b1ee9d1c0991c0f041
---
 init/Android.mk | 1 +
 init/init.cpp   | 2 +-
 2 files changed, 2 insertions(+), 1 deletion(-)

diff --git a/init/Android.mk b/init/Android.mk
index c89bacc99e..a41f669c9a 100644
--- a/init/Android.mk
+++ b/init/Android.mk
@@ -41,6 +41,7 @@ init_cflags += \
 
 include $(CLEAR_VARS)
 LOCAL_CPPFLAGS := $(init_cflags)
+LOCAL_CPPFLAGS += -DTARGET_PRODUCT=\"$(TARGET_PRODUCT)\"
 LOCAL_SRC_FILES := main.cpp
 
 LOCAL_MODULE:= init
diff --git a/init/init.cpp b/init/init.cpp
index 4fe115e924..3c020366e7 100644
--- a/init/init.cpp
+++ b/init/init.cpp
@@ -381,7 +381,7 @@ static void export_kernel_boot_props() {
         { "ro.boot.mode",       "ro.bootmode",   "unknown", },
         { "ro.boot.baseband",   "ro.baseband",   "unknown", },
         { "ro.boot.bootloader", "ro.bootloader", "unknown", },
-        { "ro.boot.hardware",   "ro.hardware",   "unknown", },
+        { "ro.boot.hardware",   "ro.hardware",   TARGET_PRODUCT, },
         { "ro.boot.revision",   "ro.revision",   "0", },
     };
     for (size_t i = 0; i < arraysize(prop_map); i++) {
-- 
2.17.1

